ballerina-test/src/test/java/org/ballerinalang/distribution/test/OpenAPIArtifactBuildTest.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.distribution.test;
+
+import org.ballerinalang.distribution.utils.TestUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.ballerinalang.distribution.utils.TestUtils.DISTRIBUTIONS_DIR;
+import static org.ballerinalang.distribution.utils.TestUtils.MAVEN_VERSION;
+import static org.ballerinalang.distribution.utils.TestUtils.RESOURCES_PATH;
+
+/**
+ * OpenAPI Tests related to artifact generation.
+ */
+public class OpenAPIArtifactBuildTest {
+    public static final String WHITESPACE_PATTERN = "\\s+";
+    @DataProvider(name = "distribution-provider")
+    public Object[][] distributionNameProvider() {
+        return new Object[][]{
+                {"ballerina-" + MAVEN_VERSION}
+        };
+    }
```

Comment:
As we are going to use only zip, IMO we don't need to have this data provider. Shall we directly define a String variable for distName?

Reply:
Here, I used data provider with the hope of adding tests for  OS wise in future. Currently, it will be better to replace with string. I will do it.   :)